skopt/searchcv.py

```diff
@@ -393,7 +393,9 @@ def add_spaces(self, names, search_spaces):
 
         if not isinstance(search_spaces, list):
             search_spaces = [search_spaces]
-        if not isinstance(names, list):
+        if isinstance(names, list):
```

Comment:
not sure this is what we really want, "if it is a list, then make a list"?

The more I think about this, can't we make our life easier by requiring that this method has only one argument `spaces` which has to be a list of `({'param1': ..., 'param2': ..}, 2)` or `{'param1': ..., 'param2': ..}` items. Internal to the method we then make up names.

Can you remind me why we need to name the spaces at all? Couldn't we switch `self.search_spaces_` to a list instead? We can check if a search space is already in the list with `space in self.search_spaces_`.

To have named spaces (and dimensions) I would go the route of adding it to `skopt.Space` etc and then taking it from there. Most `BayesSearchCV` users probably come from `sklearn.GridSearchCV` where they can't name their spaces either.